### ISSUE TITLE
Fail to build image using nodejs version is 14.17.X  for the ExpressApp project on Linux

### ISSUE THREAD

**Issue by v-xinda** [microsoft/vscode-docker#3134]
**OS:** Linux
**Build Version:** 20210809.1
**Regression:** Not a Regression

**Repro Steps:**
1. Create an ExpressApp project with following commands.(Make sure **nodejs version: 14.17.4**)
```
sudo npm install -g express-generator
express myExpressApp
cd myExpressApp
npm install
set DEBUG=myexpressapp:*
set PORT=3001
npm start
```
2. Open it in VS Code.
3. F1 -> Docker: Add Docker files to workspace -> node.js -> package.json -> 3001 -> Yes.
4. Add the following two "ENV" lines to "Dockerfile" file.
```
ENV PORT 3001
ENV DEBUG myexpressapp:*
```
5. Right click "Dockerfile" file -> Build Image.
6. Check whether succeed to build image or not.

**Expect:**
Succeed to build image.

**Actual:**
Fail to build image.
![image](https://user-images.githubusercontent.com/57662131/129164433-50cb2188-ff20-4797-ad30-d57877b44fc8.png)

**More Info:**
1. This issue also reproduces on GitHub Codespaces environment.(**nodejs version: 14.15.1**)
2. This issue doesn't reproduce with **nodejs version is 8.10.0**.

**Comment by bwateratmsft** [comment]
I haven't been able to reproduce, but I suspect it might be some transient issue with NPM:

https://github.com/snowpackjs/create-snowpack-app/issues/92

**Comment by v-xinda** [comment]
@bwateratmsft  This issue still reproduces on Linux and GitHub Codespaces environments when debugging the Node.js project. I will track this issue until it is fixed.

**Comment by bwateratmsft** [comment]
I'm able to reproduce now, taking a look.

**Comment by bwateratmsft** [comment]
@karolz-ms I'm noticing something I don't think I've seen before. The Dockerfile looks like this:

```dockerfile
FROM node:14-alpine
ENV NODE_ENV=production
USER node
WORKDIR /usr/src/app
COPY ["package.json", "package-lock.json*", "npm-shrinkwrap.json*", "./"]
RUN npm install --production --silent && mv node_modules ../
COPY . .
EXPOSE 3000
CMD ["npm", "start"]
```

If the host is Windows, `/usr/src/app` will be owned by user `node`. However, if the host is Linux, `/usr/src/app` will be owned by `root`. Any idea why the two behave differently?

**Comment by bwateratmsft** [comment]
This has to be fixed before we release 1.16.0, otherwise we will be scaffolding Dockerfiles that are inherently broken, and users will have to rescaffold later to get the fix.

**Comment by karolz-ms** [comment]
@bwateratmsft 🤷‍♂️
